components/Chat.js

```diff
@@ -99,11 +99,12 @@ export default class Chat extends React.Component {
                 _id: data._id,
```

Comment:
I checked the screenshot you submitted to your Tutor, and you don't have any `_id` field in your firestore document (you have uid instead). I think that's why it is crashing. 

I checked the exercise and they give a link to the message object structure https://github.com/FaridSafi/react-native-gifted-chat#message-object. As you can see there is no `uid` there. I suggest you simply remove uid from the documents and use `_id`

Reply:
UPDATE: I ran your code locally and debugged by putting a console log.

As you can see the entry with text `another Hello 2` is causing you troubles, since it doesn't contain a user property. Simply remove this entry from your database, uncomment the code you commented and all should work fine
<img width="1359" alt="Screenshot 2022-05-15 at 22 25 49" src="https://user-images.githubusercontent.com/2852703/168492614-38f2fb6e-a4d2-4769-8055-5f5174a4fc12.png">